### PR TITLE
Client Memory Optimizations

### PR DESCRIPTION
**Pull request by holgerd77**
Memory optimizations for the client and in general, can either be directly merged or used as a base for additional pushes, don't really care, what fits better.

### CODE REVIEW

File: packages/client/lib/execution/execution.ts [hunk 1]
```diff
@@ -56,15 +56,21 @@ export abstract class Execution {
    */
   async open(): Promise<void> {
     this.started = true
-    this.config.logger.info('Setup EVM execution.')
+    if (this.config.execution) {
```
Comment (g11tech) [inline packages/client/lib/execution/execution.ts:59]: i think we should not change the log UX here as the execution could still be triggered by beacon sync for e.g. and then stop would not log the stop message.

instead we should just add a log when we trigger execution run in full service start

File: packages/client/lib/service/fullethereumservice.ts [hunk 2]
```diff
@@ -162,7 +162,7 @@ export class FullEthereumService extends EthereumService {
     }
     await super.start()
     this.miner?.start()
-    if (!this.config.execCommon.gteHardfork(Hardfork.Paris)) {
+    if (!this.config.execCommon.gteHardfork(Hardfork.Paris) && this.config.execution) {
       void this.execution.run(true, true)
```
Comment (g11tech) [inline packages/client/lib/service/fullethereumservice.ts:166]:
```suggestion
     this.logger.info("Triggering execution of already downloaded blocks if any")
      void this.execution.run(true, true)
```
something like this...

File: packages/client/lib/execution/execution.ts (hunk 1 above)
Comment (holgerd77) [reply packages/client/lib/execution/execution.ts:59]: 👍

Could you eventually directly push ah update?

**Comment by holgerd77** [comment]
Will admin merge here for procedural reasons, we can see if codecov not reporting was eventually a one time thing on the next PR.

**Pull request merged**